src/GameStats1L.java

```diff
@@ -0,0 +1,169 @@
+ * <li>For each game {@code g} in {@code dom(GS)}, {@code GS(g)} is the sequence
+ * ⟨{@code stats.value(g).entry(0)}, …, {@code stats.value(g).entry(n‑1)}⟩ where
+ * {@code n = stats.value(g).length()}</li>
+ * </ul>
+ * </p>
+ */
+public class GameStats1L extends GameStatsSecondary {
+
+    //Representation
+
+    /**
+     * Map from game names to their logged scores.
+     * <p>
+     * The key is a non‑empty {@code String} naming the game. The value is a
+     * {@code Sequence<Integer>} that stores every score logged for that game in
+     * the exact order they were added. All scores are non‑negative.
+     * </p>
+     */
+    private Map<String, Sequence<Integer>> stats;
+
+    //Constructors
+
+    /**
+     * Default constructor: initializes an empty game‑statistics object.
+     */
+    public GameStats1L() {
+        this.createNewRep();
+    }
+
+    /**
+     * Replaces the current representation with a fresh, empty one.
+     */
+    private void createNewRep() {
+        this.stats = new Map1L<>();
+    }
+
+    //Kernel Methods
+
+    @Override
+    public void logGame(String game, int score) {
+        assert game != null : "Violation of: game is not null";
+        assert score >= 0 : "Violation of: score >= 0";
+
+        Sequence<Integer> seq;
+        if (this.stats.hasKey(game)) {
+            seq = this.stats.value(game);
+        } else {
+            seq = new Sequence1L<>();
+            this.stats.add(game, seq);
+        }
+        seq.add(seq.length(), score);
+    }
+
+    @Override
+    public int getBestScore(String game) {
+        assert game != null : "Violation of: game is not null";
+
+        if (!this.stats.hasKey(game) || this.stats.value(game).length() == 0) {
+            return -1;
+        }
+        Sequence<Integer> seq = this.stats.value(game);
+        int best = seq.entry(0);
+        for (int i = 1; i < seq.length(); i++) {
+            int s = seq.entry(i);
+            if (s > best) {
+                best = s;
+            }
+        }
+        return best;
+    }
+
+    @Override
+    public int getTotalGamesPlayed(String game) {
+        assert game != null : "Violation of: game is not null";
+        if (this.stats.hasKey(game)) {
+            return this.stats.value(game).length();
+        } else {
+            return 0;
+        }
+    }
+
+    //Standard methods
+
+    @Override
+    public void clear() {
+        this.createNewRep();
+    }
+
+    @Override
+    public GameStats1L newInstance() {
+        return new GameStats1L();
+    }
+
+    @Override
+    public void transferFrom(GameStats source) {
+        assert source != null : "Violation of: source is not null";
+        assert source != this : "Violation of: source is not this";
+
+        GameStats1L src = (GameStats1L) source;
+        this.stats = src.stats;
+        src.createNewRep();
+    }
+
+    //Methods required by GameStatsSecondary
+
+    @Override
+    protected int getScoreAt(String game, int index) {
```

Comment:
This part is a little strange. Having private methods is fine, but I'm not sure why these are here if they're not kernels. Either these are kernel methods or they're secondary methods. If they're secondary, they should be moved. If they're kernels, they should be included in the kernel interface.